### PR TITLE
fix showing NotFound page when offline

### PR DESCRIPTION
**Pull request by hungvu193**
### Details

Fix showing NotFound page when offline
### Fixed Issues

$ https://github.com/Expensify/App/issues/30246
PROPOSAL: https://github.com/Expensify/App/issues/30246#issuecomment-1778492320

### Tests

1. Log in to any Account.
2. Open any chat and select a Contact to view their details.
3. Select any private notes and copy the link, paste it to the chat.
4. Logout and login again, disable the internet connection, now open the chat and click on the private notes deeplinking.
5. Turn on the internet connection.
6. Notice that it loads the private notes.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as Tests.
### QA Steps

1. Log in to any Account.
2. Open any chat and select a Contact to view their details.
3. Select any private notes and copy the link, paste it to the chat.
4. Logout and login again, disable the internet connection, now open the chat and click on the private notes deeplinking.
5. Turn on the internet connection.
6. Notice that it loads the private notes.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/16502320/d5157536-672b-45c9-8f67-0e86d4871fdd

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/16502320/c6bfbc4d-668d-4df7-9ead-e37e05b16628

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by hungvu193** [comment]
@jjcoffee PR is ready! I had a little trouble with my internet connection, please review the PR I'll complete the screenshot later.

**Comment by jjcoffee** [comment]
@hungvu193 Are you able to add the rest of the screenshots?

**Comment by jjcoffee** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by jjcoffee** [comment]
@hungvu193 There's no back button, so no (in-app) way to navigate back on mobile:

https://github.com/Expensify/App/assets/27287420/09770c88-f7e2-442d-9bae-c5ff91a20924

**Comment by jjcoffee** [comment]
Found a bit of an edge case where the not found error still shows, I think it's if you go offline just as the `GetReportPrivateNotes` requests starts. Any way to handle it?

image

**Comment by hungvu193** [comment]
let me take a look. I didn't see it before

> Found a bit of an edge case where the not found error still shows, I think it's if you go offline just as the `GetReportPrivateNotes` requests starts. Any way to handle it?
> 
>  desktop-app-2023-11-17_16.20.27.mp4 
> image
> 
> The screen is then just blank when I come back online:
> 
> image

**Comment by hungvu193** [comment]
I couldn't reproduce it, can you provide the Tests steps? @jjcoffee

**Comment by jjcoffee** [comment]
@hungvu193 If you go offline whilst the `GetReportPrivateNotes` is still pending (switching the network to slow 3G helps with this), it happens pretty consistently.

Can you also look into this [issue](https://github.com/Expensify/App/pull/31257#issuecomment-1816588877)? Thanks!

**Comment by hungvu193** [comment]
> @hungvu193 There's no back button, so no (in-app) way to navigate back on mobile:
> 
>  android-chrome-2023-11-17_16.03.16.mp4

Since we moved the FullScreenLoadingIndicator to `withReportAndPrivateNotesOrNotFound` to reuse the logic, I think it's expected (it was added by another PR, so I think it's out of scope in this PR).

**Comment by jjcoffee** [comment]
@hungvu193 Sorry, but I think we need to implement a correct fix here, regardless of what the other PR has implemented (the way the loading screen displays should have been a regression IMO). 

Based on your [proposal](https://github.com/Expensify/App/issues/30246#issuecomment-1778492320), we would have shown the loading indicator within the screen, below the header, so I think let's go ahead and implement that here and keep bad UX out of the app :smile:

**Comment by hungvu193** [comment]
> @hungvu193 Sorry, but I think we need to implement a correct fix here, regardless of what the other PR has implemented (the way the loading screen displays should have been a regression IMO).
> 
> Based on your [proposal](https://github.com/Expensify/App/issues/30246#issuecomment-1778492320), we would have shown the loading indicator within the screen, below the header, so I think let's go ahead and implement that here and keep bad UX out of the app 😄

I think that's ok to fix it, however other HOCs also used the same approach by using FullScreenLoadingIndicator inside.
Do you think we really to fix it?
Also if we want to fix it, I think I'll wrap our FullScreenLoadingIndicator inside a View that has Header. How do you think?

**Comment by jjcoffee** [comment]
@hungvu193 Can you give some examples where other HOCs do the same thing?

**Comment by hungvu193** [comment]
> @hungvu193 Can you give some examples where other HOCs do the same thing?

Yeah
https://github.com/Expensify/App/blob/72e73ae6d922b915198e0a5bcff9d71de5a1001b/src/pages/home/report/withReportOrNotFound.tsx#L50-L56
and
https://github.com/Expensify/App/blob/72e73ae6d922b915198e0a5bcff9d71de5a1001b/src/pages/home/report/withReportAndReportActionOrNotFound.js#L109-L117

**Comment by jjcoffee** [comment]
Are there some steps to get one of these to show? I'm curious where this pattern actually shows up on the FE.

**Comment by jjcoffee** [comment]
Also any thoughts on the [issue](https://github.com/Expensify/App/pull/31257#issuecomment-1818763361) when going offline?

**Comment by jjcoffee** [comment]
@hungvu193 Friendly bump on the [above](https://github.com/Expensify/App/pull/31257#issuecomment-1820993077).

**Comment by hungvu193** [comment]
> @hungvu193 Friendly bump on the [above](https://github.com/Expensify/App/pull/31257#issuecomment-1820993077).

Oops sorry, I've been so busy on my jobs. I'll give you a solution tomorrow

**Comment by jjcoffee** [comment]
@hungvu193 Any update on this?

**Comment by hungvu193** [comment]
Oh sorry! I've been out sick last couple of weeks and can't be available next few days. Do you have any suggesstion for the solutions? I'll take a look when I feel better.

**Comment by hungvu193** [comment]
@jjcoffee So in the edge case, the response is failed and we will show the NotFoundPage, also if you reconnected to the internet it will redirect you to add a note. Is that ok? 

https://github.com/Expensify/App/assets/16502320/1f365d94-1b27-40ae-9812-155171bd533e

**Comment by hungvu193** [comment]
@jjcoffee I updated the code!
The idea here is that we will show the loading even the api request was failed if that's our private notes and it's empty.

**Comment by jjcoffee** [comment]
@hungvu193 Thanks for the update, hope you're feeling better! On testing the same scenario on a report with some note saved, I am now seeing that it just loads the notes page without any notes (no loader). On going back online the notes don't appear. Can you take a look?

**Comment by jjcoffee** [comment]
@hungvu193 Could you also look at getting the loader to display below the header as you mention [here](https://github.com/Expensify/App/pull/31257#issuecomment-1820200256)?

**Comment by hungvu193** [comment]
> @hungvu193 Thanks for the update, hope you're feeling better! On testing the same scenario on a report with some note saved, I am now seeing that it just loads the notes page without any notes (no loader). On going back online the notes don't appear. Can you take a look?

That's weird. Let me see

**Comment by hungvu193** [comment]
> @hungvu193 Could you also look at getting the loader to display below the header as you mention [here](https://github.com/Expensify/App/pull/31257#issuecomment-1820200256)?

Sure, I'll handle it similar with FullpageNotfoundView.
Also can you record a video about the bug you mentioned above? I can't seem to reproduce

**Comment by hungvu193** [comment]
I added a LoadingPage, here's how it looks. At the moment I only applied it for `withReportAndPrivateNotesOrNotFound` HOC, let me know if you want to apply it for another HOCs as well.

https://github.com/Expensify/App/assets/16502320/eb72ce01-8f99-431a-af46-f08d4e52b055

**Comment by jjcoffee** [comment]
Any way we can get the header to show as well?

**Comment by jjcoffee** [comment]
Sorry for not being clearer! I meant the header text:

image

**Comment by hungvu193** [comment]
> Sorry for not being clearer! I meant the header text:
> 
> 
> 
> image
> 
> 

Ah I see, We can do that by passing extra props to our HOCs, however with NotFoundPage, we also didn't show the header text, so I don't think it's necessary for this case 😃

**Comment by jjcoffee** [comment]
Hmm I think I prefer the way it worked on [this PR ](https://github.com/Expensify/App/pull/27911) that we worked on together a while ago :smile: 

With the the not found view, it makes sense (in a way) since the only action for the user is to go back, so it doesn't really matter to display a header. However, for the loading case the header will "suddenly" appear once the notes finish loading/you come back online. So, if it's not too complex to implement I'd go for it!

**Comment by hungvu193** [comment]
> Hmm I think I prefer the way it worked on [this PR ](https://github.com/Expensify/App/pull/27911) that we worked on together a while ago :smile: 
> 
> 
> 
> With the the not found view, it makes sense (in a way) since the only action for the user is to go back, so it doesn't really matter to display a header. However, for the loading case the header will "suddenly" appear once the notes finish loading/you come back online. So, if it's not too complex to implement I'd go for it!

Makes sense 😄. I'll take a look. Is there anything else so we can push this forward ⚡️

**Comment by hungvu193** [comment]
@jjcoffee I updated the code, now we can use `withReportAndPrivateNotesOrNotFound('pageTitle')`, with `pageTitle` is the title of the loading page.

https://github.com/Expensify/App/assets/16502320/a0d5cd66-b797-4fa5-82e5-a4be0a3ef075

**Comment by jjcoffee** [comment]
Hmm this doesn't seem very DRY, since we're now just (partially) replicating `PrivateNotesListPage`'s layout in `LoadingPage`. I'd think it might make sense to move the `FullScreenLoadingIndicator` back to the `PrivateNotesListPage` component? I'm a bit weak on HOCs though, so I might be missing some obvious reason why that's dumb :smile:

**Comment by hungvu193** [comment]
> Hmm this doesn't seem very DRY, since we're now just (partially) replicating `PrivateNotesListPage`'s layout in `LoadingPage`. I'd think it might make sense to move the `FullScreenLoadingIndicator` back to the `PrivateNotesListPage` component? I'm a bit weak on HOCs though, so I might be missing some obvious reason why that's dumb :smile:

I think the reason of using that HOC to reuse the logic private note, that's why. If we bring back the FullscreenIndicator, the logic of that HOCs will be redundant 😬.
Also our NotfoundPage used the same way with our current LoadingPage, so I think that's ok to make it this way. 🤣

**Comment by jjcoffee** [comment]
That makes sense, thanks! 

So then I think the final thing to deal with here is the issue I mentioned previously, here's a video demonstrating it:

https://github.com/Expensify/App/assets/27287420/36b1e6f7-07f0-44ff-bec4-37fbf30fd562

The key is to go offline before the `GetReportPrivateNote` request completes. Using a throttled connection helps with that.

**Comment by hungvu193** [comment]
> That makes sense, thanks! 
> 
> 
> 
> So then I think the final thing to deal with here is the issue I mentioned previously, here's a video demonstrating it:
> 
> 
> 
> https://github.com/Expensify/App/assets/27287420/280c3991-4896-467c-9810-8a7dd5a1bcc7
> 
> 
> 
> The key is to go offline before the `GetReportPrivateNote` request completes. Using a throttled connection helps with that.
> 
> 

I see 😮‍💨 . Seems some logics were added in the meantime that will navigate the screen to edit note page if our notes are failed to load. Let me take a look.
Also can you reproduce this case on main? If that we can consider it a regression of the current implementation logic.

**Comment by jjcoffee** [comment]
On main I get a not found page under the same circumstance, so I think it's within scope to fix.

**Comment by hungvu193** [comment]
> On main I get a not found page under the same circumstance, so I think it's within scope to fix.

Thanks for testing it. I'm on it 🫡

**Comment by hungvu193** [comment]
@jjcoffee I believe these lines were added weeks ago caused the issue you mentioned.
https://github.com/Expensify/App/blob/78ad3f9047025a37382a231a1968211c8ad4685a/src/libs/ReportUtils.ts#L4310-L4314

It's checking if the note is empty then navigating to edit note page. So I think it's kind of out of scope for this PR.

File: src/pages/LoadingPage.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                onBackButtonPress={props.onBackButtonPress}
+                shouldShowBackButton
+                title={props.title}
+            />
+            <FullScreenLoadingIndicator style={[styles.flex1, styles.pRelative]} />
+        </ScreenWrapper>
+    );
+}
+
+LoadingPage.displayName = 'NotFoundPage';
```
Comment (jjcoffee) [inline src/pages/LoadingPage.js:33]:
```suggestion
LoadingPage.displayName = 'LoadingPage';
```
:wink:

**Review by jjcoffee — APPROVED** [review]
Tests well and LGTM!

**Comment by jjcoffee** [comment]
Reported the [issue](https://github.com/Expensify/App/pull/31257#issuecomment-1855876502) when the request doesn't complete [here](https://github.com/Expensify/App/issues/32302#issuecomment-1856202756).

**Comment by hungvu193** [comment]
Friendly bump @puneetlath

**Comment by puneetlath** [comment]
Sorry to be so late on this. Can you explain why we are creating a new loading page instead of using the `FullPageNotFoundView`? I couldn't find that in the proposal.

**Comment by hungvu193** [comment]
> Sorry to be so late on this. Can you explain why we are creating a new loading page instead of using the `FullPageNotFoundView`? I couldn't find that in the proposal.

@puneetlath I think you mean `FullpageLoadingIndicator`, it will display fullscreen page without Header, so we create a new one, similar to `FullpageNotFoundView` so we can see the Header.

**Comment by jjcoffee** [comment]
@puneetlath There's a bit more context [here](https://github.com/Expensify/App/pull/31257#issuecomment-1816588877) - it's mainly useful to have the header display on mobile as otherwise there's no way to navigate back from the loading screen.

File: src/pages/LoadingPage.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    /** Method to trigger when pressing back button of the header */
+    onBackButtonPress: PropTypes.func,
+    title: PropTypes.string.isRequired,
+};
+
+const defaultProps = {
+    onBackButtonPress: undefined,
+};
+
+// eslint-disable-next-line rulesdir/no-negated-variables
```
Comment (puneetlath) [inline src/pages/LoadingPage.js:18]: Why is this needed?

**Review by puneetlath — CHANGES_REQUESTED** [review]
One small comment. Otherwise seems good to me.

**Comment by puneetlath** [comment]
@jjcoffee the reviewer checklist check is also failing.

**Comment by jjcoffee** [comment]
Checklist updated to the latest version!

**Comment by puneetlath** [comment]
Ok great. Just one outstanding question for you then @hungvu193.

File: src/pages/LoadingPage.js (hunk 2 above)
Comment (hungvu193) [reply src/pages/LoadingPage.js]: I see. It seems redundant. Updated!

**Review by puneetlath — APPROVED** [review]

**Comment by stitesExpensify** [comment]
:wave: I think this PR may have caused this deploy blocker https://github.com/Expensify/App/issues/34270. Would you mind taking a look @jjcoffee @puneetlath ?

File: src/pages/home/report/withReportAndPrivateNotesOrNotFound.js
Comment (sibtain-ali-deel) [inline src/pages/home/report/withReportAndPrivateNotesOrNotFound.js:1]: We are not showing an offline page when the user is not connected to the internet. This caused https://github.com/Expensify/App/issues/55303

Comment (allroundexperts) [inline src/pages/home/report/withReportAndPrivateNotesOrNotFound.js:1]: We are not showing an offline page when the user is not connected to the internet. This caused https://github.com/Expensify/App/issues/55303

**Pull request merged**